packages/components/src/utils/property-checkers/index.ts

```diff
@@ -1,15 +1,14 @@
-import { emptyOr } from './empty-or';
+import { requiredAnd } from './required-and';
 import { checkOneOf } from './check-one-of';
 import { checkPattern } from './check-pattern';
 import { checkType } from './check-type';
 import { checkUrl } from './check-url';
 
-export const checkEmptyOrOneOf = emptyOr(checkOneOf);
-export const checkEmptyOrPattern = emptyOr(checkPattern);
-export const checkEmptyOrType = emptyOr(checkType);
-export const checkEmptyOrUrl = emptyOr(checkUrl);
+export const checkRequiredAndOneOf = requiredAnd(checkOneOf);
```

Comment:
Currently, the `isValueEmpty()` run twice in the `checkRequiredAnd*` functions: first in the `requiredAnd` and then in the checker itself. 

What you could do is something like: 
```typescript
import { requiredAnd } from './required-and';
import { emptyOr } from './empty-or';
import { patternChecker } from './pattern-checker';

export const checkPattern = emptyOr(patternChecker);
export const checkRequiredAndPattern = requiredAnd(patternChecker);
```